diff --git a/hiredis.h b/hiredis.h
index f28dcdb..e120cb5 100644
--- a/hiredis.h
+++ b/hiredis.h
@@ -30,6 +30,11 @@
 #ifndef __HIREDIS_H
 #define __HIREDIS_H
 
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
 #define REDIS_REPLY_ERROR 0
 #define REDIS_REPLY_STRING 1
 #define REDIS_REPLY_ARRAY 2
@@ -51,4 +56,8 @@ redisReply *redisConnect(int *fd, const char *ip, int port);
 void freeReplyObject(redisReply *r);
 redisReply *redisCommand(int fd, const char *format, ...);
 
+#ifdef __cplusplus
+}
+#endif
+
 #endif

